### ISSUE TITLE
ClientSession.get is timing out

### ISSUE THREAD

**Issue by srob1** [aio-libs/aiohttp#5467]
🐞 **Describe the bug**

My call to `ClientSession.get()` for the url 'https://www.accuweather.com/en/us/new-york/10007/current-weather/349727' times out.  The same url passed to `requests.get()` works fine.

💡 **To Reproduce**

Here is the aiohttp version that times out:

```python
#!/usr/bin/env python
#coding=utf-8

import aiohttp
import asyncio

HEADERS = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.104 Safari/537.36',
}

async def main():
    timeout=aiohttp.ClientTimeout(total=30)
    url = 'https://www.accuweather.com/en/us/new-york/10007/current-weather/349727'
    async with aiohttp.ClientSession(timeout=timeout) as session:
        async with session.get(url, headers=HEADERS) as response:
            text = await response.text(encoding='utf-8')
    print(len(text))

if __name__ == '__main__':
    asyncio.run(main())
```

Here is the equivalent blocking version which works fine:

```python
#!/usr/bin/env python
#coding=utf-8

import requests

HEADERS = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.104 Safari/537.36',
}

def main():
    url = 'https://www.accuweather.com/en/us/new-york/10007/current-weather/349727'
    page = requests.get(url, headers=HEADERS)
    text = page.text
    print(len(text))

if __name__ == '__main__':
    main()
```

💡 **Expected behavior**

ClientSession.get() shouldn't be timing out.  It should be able to retrieve the requested web page just like the requests.get() version.  Without the 30-second timeout, it still times out, it just takes 5 minutes (or whatever the default timeout is).

📋 **Logs/tracebacks**

Here is a traceback of the timeout from the aiohttp version:
```python-traceback (paste your traceback in the next line)
> test1.py
Traceback (most recent call last):
  File "c:\My Documents\test1.py", line 20, in <module>
    asyncio.run(main())
  File "c:\users\user\appdata\local\programs\python\python38\lib\asyncio\runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "c:\users\user\appdata\local\programs\python\python38\lib\asyncio\base_events.py", line 616, in run_until_complete
    return future.result()
  File "c:\My Documents\test1.py", line 15, in main
    async with session.get(url, headers=HEADERS) as response:
  File "c:\Users\user\Envs\default\lib\site-packages\aiohttp\client.py", line 1117, in __aenter__
    self._resp = await self._coro
  File "c:\Users\user\Envs\default\lib\site-packages\aiohttp\client.py", line 544, in _request
    await resp.start(conn)
  File "c:\Users\user\Envs\default\lib\site-packages\aiohttp\client_reqrep.py", line 905, in start
    self._continue = None
  File "c:\Users\user\Envs\default\lib\site-packages\aiohttp\helpers.py", line 656, in __exit__
    raise asyncio.TimeoutError from None
asyncio.exceptions.TimeoutError
Exception ignored in: <function _ProactorBasePipeTransport.__del__ at 0x000002A00B218430>
Traceback (most recent call last):
  File "c:\users\user\appdata\local\programs\python\python38\lib\asyncio\proactor_events.py", line 116, in __del__
    self.close()
  File "c:\users\user\appdata\local\programs\python\python38\lib\asyncio\proactor_events.py", line 108, in close
    self._loop.call_soon(self._call_connection_lost, None)
  File "c:\users\user\appdata\local\programs\python\python38\lib\asyncio\base_events.py", line 719, in call_soon
    self._check_closed()
  File "c:\users\user\appdata\local\programs\python\python38\lib\asyncio\base_events.py", line 508, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
```

Here is the output from the blocking version that uses requests.get:
```console
> test2.py
99906

(Although the length printed can vary as the page content changes.)
```

📋 **Your version of the Python**

```console
$ python --version
Python 3.8.3
```

📋 **Your version of the aiohttp/yarl/multidict distributions**

```console
$ python -m pip show aiohttp
Name: aiohttp
Version: 3.7.3
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: Nikolay Kim
Author-email: fafhrd91@gmail.com
License: Apache 2
Location: c:\users\user\envs\default\lib\site-packages
Requires: chardet, multidict, attrs, typing-extensions, yarl, async-timeout
Required-by:
```
```console
$ python -m pip show multidict
Name: multidict
Version: 5.1.0
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: c:\users\user\envs\default\lib\site-packages
Requires:
Required-by: yarl, aiohttp
```
```console
$ python -m pip show yarl
Name: yarl
Version: 1.6.3
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: c:\users\user\envs\default\lib\site-packages
Requires: multidict, idna
Required-by: aiohttp
```

📋 **Additional context**

client

**Comment by webknjaz** [comment]
I've checked that URL with cURL and noticed that it immediately switches over to HTTP/2, then spits out the response.

But when I force HTTP/1.1, it gets stuck after the TLS handshake with nothing being output:
```
$ curl -v -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.104 Safari/537.36' https://www.accuweather.com/en/us/new-york/10007/current-weather/349727 --http1.1
```

Looking at the Requests III docs at https://3.python-requests.org/, it looks like they have HTTP/2 support in WIP and that's why you've been able to get a result.

aiohttp does not support HTTP/2 at this time, nor there is any effort to change this. There's been some discussions related to this in the past like https://github.com/aio-libs/aiohttp/issues/320 / https://github.com/aio-libs/aiohttp/issues/863 but it never got enough attention or interest.
That said if you want to explore contributing HTTP/2 support to aiohttp, feel free to do so and send a PR. But I don't think any of the core maintainers have the resources to work on this feature.

Looking at the ecosystem, it seems like https://pypi.org/p/httpcore / https://pypi.org/p/httpx support this.

Ref: https://bugs.python.org/issue23794

**Comment by srob1** [comment]
Thank you for taking the time to research my issue so quickly and for all the helpful information, particularly the pointers to `httpcore` and `httpx`.  I changed my code to use `httpx` and it's working, so I definitely appreciate that.

Since this issue arose in the context of writing my very first `asyncio`/`aiohttp` python app, contributing `HTTP/2` support to `aiohttp` is way beyond my current abilities.  Although I hope someone gets interested in adding `HTTP/2` and `HTTP/3` support, as I think they are the future.

Thank you again!

**Comment by Dreamsorcerer** [comment]
While implementing HTTP/2 is a big task, we should probably atleast detect that it is HTTP/2 or something else we can't handle and give a proper error message here. Having the connection just hang is not very helpful for debugging.

**Comment by webknjaz** [comment]
@Dreamsorcerer I think that in this case, it's a server misconfiguration: the severs _suggests_ upgrading to HTTP/2 but an HTTP/1.1 client should just ignore that and keep talking HTTP/1.1 which it does but the server doesn't want to proceed and keeps the TCP connection hanging w/o any indication that it doesn't want HTTP/1.1 or anything. It doesn't terminate the connection and there's no way for us to know if that's on purpose or the server is just damn slow/overloaded.

As demonstrated above, with cURL that 100% supports HTTP/2, if I force it to use HTTP/1.1 on that host, it just hangs the same way.

The bottom line is that the host **`accuweather.com` is broken**, not us. If you do the came cURL test with `https://google.com`, it will keep talking HTTP/1.1 if forced to do so.

**Comment by Dreamsorcerer** [comment]
So, actually, the server simply doesn't respond at all unless you include the Connection: Upgrade header. If you include that, then it'll give a valid HTTP/1.1 response and aiohttp works just fine.

So, just adding into the original example, in HEADERS:
```
    'Connection': 'Upgrade',
```

It then works fine. So, the server is just a bit broken and I don't think there's anything to fix here.
